pkg/k8s/policy.go

```diff
@@ -55,47 +52,85 @@ func ServerAuthorizationsForResource(ctx context.Context, k8sAPI *KubernetesAPI,
 			selector, err := metav1.LabelSelectorAsSelector(saz.Spec.Server.Selector)
 			if err != nil {
-				fmt.Fprintf(os.Stderr, "Failed to get servers: %s\n", err)
-				os.Exit(1)
+				fmt.Fprintf(os.Stderr, "Failed to parse Server selector for ServerAuthorization/%s: %s\n", saz.Name, err)
+				continue
 			}
 			serverList, err := k8sAPI.L5dCrdClient.ServerV1beta1().Servers(saz.GetNamespace()).List(ctx, metav1.ListOptions{LabelSelector: selector.String()})
 			if err != nil {
-				fmt.Fprintf(os.Stderr, "Failed to get servers: %s\n", err)
-				os.Exit(1)
+				fmt.Fprintf(os.Stderr, "Failed to get Servers for ServerAuthorization/%s: %s\n", saz.Name, err)
+				continue
 			}
 			servers = serverList.Items
 		}
 
 		for _, server := range servers {
-			if server.Spec.PodSelector == nil {
-				continue
+			if serverIncludesPod(server, pods) {
+				results = append(results, Authorization{
+					Route:               "",
+					Server:              server.GetName(),
+					ServerAuthorization: saz.GetName(),
+					AuthorizationPolicy: "",
+				})
 			}
+		}
+	}
+
+	policies, err := k8sAPI.L5dCrdClient.PolicyV1alpha1().AuthorizationPolicies(namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to get AuthorizationPolicy resources: %s\n", err)
+		os.Exit(1)
+	}
 
-			selector, err := metav1.LabelSelectorAsSelector(server.Spec.PodSelector)
+	for _, p := range policies.Items {
+		target := p.Spec.TargetRef
+		if target.Kind == ServerKind && target.Group == PolicyAPIGroup {
+			server, err := k8sAPI.L5dCrdClient.ServerV1beta1().Servers(p.Namespace).Get(ctx, string(target.Name), metav1.GetOptions{})
 			if err != nil {
-				fmt.Fprintf(os.Stderr, "Failed to create selector: %s\n", err)
-				os.Exit(1)
+				fmt.Fprintf(os.Stderr, "AuthorizationPolicy/%s targets Server/%s but we failed get it: %s\n", p.Name, target.Name, err)
+				continue
 			}
-
-			var selectedPods []corev1.Pod
-			for _, pod := range pods {
-				if selector.Matches(labels.Set(pod.Labels)) {
-					selectedPods = append(selectedPods, pod)
-				}
+			if serverIncludesPod(*server, pods) {
+				results = append(results, Authorization{
+					Route:               "",
+					Server:              server.GetName(),
+					ServerAuthorization: "",
+					AuthorizationPolicy: p.GetName(),
+				})
 			}
-
-			if serverIncludesPod(server, selectedPods, podSet) {
-				results = append(results, ServerAndAuthorization{server.GetName(), saz.GetName()})
+		} else if target.Kind == HTTPRouteKind && target.Group == PolicyAPIGroup {
+			route, err := k8sAPI.L5dCrdClient.PolicyV1alpha1().HTTPRoutes(p.Namespace).Get(ctx, string(target.Name), metav1.GetOptions{})
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "AuthorizationPolicy/%s targets HTTPRoute/%s but we failed get it: %s\n", p.Name, target.Name, err)
+				continue
+			}
+			for _, parent := range route.Spec.ParentRefs {
+				if parent.Kind != nil && *parent.Kind == ServerKind &&
```

Comment:
It'd be nice if we add the accessors here for `GetKind()` and `GetGroup`.

Is there a reason we have the `nil` checks for `parent` but not above for `target`?

Reply:
ParentRef comes from the gateway-api, so we can't add accessors directly onto it, unfortunately.  ParentRef.Kind is a pointer (and thus needs nil checks) while PolicyTargetRef.Kind is not a pointer.